autogpt/core/messaging/base.py

```diff
@@ -1,9 +1,47 @@
-import abc
+from abc import ABC, abstractmethod
+from enum import Enum, StrEnum
+import dataclasses
+from typing import Any, List, Callable, Union
 
+class MessageCategory(Enum):
```

Comment:
The distinction between message category and role seems fuzzy.  Ie can a user only send `USER_INPUT` messages?  can the SYSTEM/ASSISTANT send anything but `USER_PROMPT`?